specs/logging_implementation.adoc

```diff
@@ -0,0 +1,202 @@
+  alert sockets etc. depending upon the metadata and configuration.
+* Syslog must be the default logging configuration. Logging to individual
+  configuration files should be thoroughly discouraged.
+
+== Alternatives
+
+* None
+
+== Data model impact
+
+None
+
+== Impacted modules
+
+=== Tendrl/common impact
+* Create a new class called "Message" in common for message handling with attributes
+  handler_name, message_type, message_id, log_level, message_tag, message, severity,
+  timestamp. (other attributes are filename, function_name, line_number, timestamp.
+  These details are appended with message automatically).
+```
+Message class object structure
+     log_handler_name
+     message_type(log/alert/flight operation)
+     log_level(info/error..)
+     message
+     message_trace_back_flag(True/False)
+     message_uuid (for log flight operation job id is uuid, for alert alert_id is uuid
+                   for normal log messages this field is optional)
+     severity(optional for log and flight operation)
+     tag(optional for log and flight operation)
+
+     timestamp(generated and appended by handler automatically)
+    (Before log any message filename, function_name, line_number appended with message
+    by handler).
+
+    1. For log - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag)
+    2. For flight operation - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag, message_uuid(job_id))
+    3. For alert - message_handler(log_handler_name, message_type, log_level, message,
+       message_trace_back_flag, message-uuid(alert_id), severity, tag)
+```
+* Create a new function called message_handler in Message class and process the
+  object values.
+* If the message type is Alert then store the message in log file with uuid (log level
+  for alert is always warning). and put the object in node_agent socket.
+* If the message type is flight operation then log the message as INFO with message
+  uuid (uuid is used to trace the filght operation).
+* If the type is Log then store the message in log file based on log level.
+
+=== Tendrl/node_agent impact
+
+* Create socket to receive alert.
+* Import message class from common and change all log statement (it will be in another PR).
+
+=== Tendrl/gluster_integration impact
+
+* Import message class from common and change all log statement (it will be in another PR).
+
+=== Tendrl/ceph_integration impact
+
+* Import message class from common and change all log statement (it will be in another PR).
+
+=== Tendrl/alerting impact
+
+* Import message class from common and change all log statement (it will be inanother PR).
+
+=== Tendrl/performance_monitoring impact
+
+* Import message class from common and change all log statement (it will be in another PR).
+
+== Security impact
+
+None
+
+== Notifications/Monitoring impact
+
+None
```

Comment:
We should probably say:

The collectd plugin to handle collectd detected threshold breaches should be customized to call the utility function described in this adoc instead of .directly posting the alert to socket